src/Microsoft.IdentityModel.Tokens/InMemoryCryptoProviderCache.cs

```diff
@@ -184,24 +259,157 @@ public override bool TryRemove(SignatureProvider signatureProvider)
                 return false;
 
             var cacheKey = GetCacheKey(signatureProvider);
+#if NETSTANDARD2_0
+            MemoryCache signatureProviderCache;
+#elif NET45 || NET461 || NET472
+            EventBasedLRUCache<string, SignatureProvider> signatureProviderCache;
+#endif
+            // Determine if we are caching a signing or a verifying SignatureProvider.
             if (signatureProvider.WillCreateSignatures)
+                signatureProviderCache = _signingSignatureProviders;
+            else
+                signatureProviderCache = _verifyingSignatureProviders;
+
+            try
             {
-                if (_signingSignatureProviders.TryRemove(cacheKey, out SignatureProvider provider))
+#if NET45 || NET461 || NET472
+                if (signatureProviderCache.TryRemove(cacheKey, out SignatureProvider provider))
                 {
                     provider.CryptoProviderCache = null;
                     return true;
                 }
-            }
-            else
-            {
-                if (_verifyingSignatureProviders.TryRemove(cacheKey, out SignatureProvider provider))
+                else
+                {
+                    return false; 
+                }
+#elif NETSTANDARD2_0
+                if (signatureProviderCache.TryGetValue(cacheKey, out SignatureProvider provider))
                 {
+                    signatureProviderCache.Remove(cacheKey);
                     provider.CryptoProviderCache = null;
                     return true;
                 }
+                else
+                {
+                    return false;
+                }
+#endif
             }
+            catch (Exception ex)
+            {
+                LogHelper.LogWarning(LogHelper.FormatInvariant(LogMessages.IDX10699, cacheKey, ex));
+                return false;
+            }
+        }
 
-            return false;
+        /// <summary>
+        /// Calls <see cref="Dispose(bool)"/> and <see cref="GC.SuppressFinalize"/>
+        /// </summary>
+        public void Dispose()
+        {
+            // Dispose of unmanaged resources.
+            Dispose(true);
+            // Suppress finalization.
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// If <paramref name="disposing"/> is true, this method disposes of <see cref="_signingSignatureProviders"/> and <see cref="_verifyingSignatureProviders"/>.
+        /// </summary>
+        /// <param name="disposing">True if called from the <see cref="Dispose()"/> method, false otherwise.</param>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!_disposed)
+            {
+                _disposed = true;
+                if (disposing)
+                {
+                    _signingSignatureProviders.Dispose();
+                    _verifyingSignatureProviders.Dispose();
+                }
+            }
+        }
+
+#region FOR TESTING TO BE REMOVED OR INTERNAL BEFORE RELEASE
```

Comment:
how do we want to resolve these before merging in?

Reply:
I think it makes sense to make these properties internal, since we'd likely want to be able to run cache performance tests even after the PR goes in.

@brentschmaltz Can the performance tests be modified to use internal rather than public properties?